msteams-platform/sbs-create-dashboard-widget-graph-api-call-in-Teams-toolkit.yml

```diff
@@ -0,0 +1,280 @@
+
+        ```
+        {
+          "content": "Hello world!"
+        }
+        ```
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/create-sample-data-json.png" alt-text="Screenshot shows the creation of sampleData.json file under the data folder in Visual Studio Code.":::
+
+     **To retrieve services**:
+
+     1. In Visual Studio Code, under **EXPLORER**, go to EXPLORER> tabs > src > data> services.
+     1. Create a **sampleService.ts** file.
+     1. Use the following code in the **sampleService.ts**:
+
+        ```
+        import { SampleModel } from "../models/sampleModel";
+        import SampleData from "../data/SampleData.json";
+
+        export const getSampleData = (): SampleModel => SampleData;
+        ```
+
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/service-retrieve.png" alt-text="Screenshot shows the creation of sampleService.ts file under the service folder in Visual Studio Code.":::
+
+      > [!NOTE]
+      > You can implement a service to retrieve data from the backend service or from Microsoft Graph API.
+    
+     ## Create a widget file
+
+     1. In Visual Studio Code, go to EXPLORER> tabs> src> >views >widgets.
+     1. Create a **sampleWidget.ts** file.
+     1. Use the following code in the **sampleWidget.tsx** file:
+
+        ```
+        import { Button, Text } from "@fluentui/react-components";
+        import { Widget } from "../lib/Widget";
+        import { SampleModel } from "../../models/sampleModel";
+        import { getSampleData } from "../../services/sampleService";
+
+        export class SampleWidget extends Widget<SampleModel> {
+        async getData(): Promise<SampleModel> {
+        return getSampleData();
+        }
+
+        headerContent(): JSX.Element | undefined {
+        return <Text>Sample Widget</Text>;
+        }
+
+        bodyContent(): JSX.Element | undefined {
+        return <div>{this.state.data?content}</div>;
+        }
+
+        footerContent(): JSX.Element | undefined {
+        return (<Button
+          appearance="primary"
+          size="medium"
+          style={{ width: "fit-content" }}
+          onClick={() => {}}
+        >
+          View Details
+        </Button>);
+        }
+        }
+  
+        ```
+        :::image type="content" source="~/assets/images/sbs-create-a-new-dashboard/create-sample-widget.png" alt-text="Screenshot shows the creation of sampleWidget.tsx under the widgets in Visual Studio Code.":::
+      
+     
+     ## Add the widget to the dashboard
+
+      If you want to create a new dashboard, refer to [add a new dashboard](tabs/how-to/dashboard-teams-toolkit.md#add-a-new-dashboard).
+
+      1. In Visual Studio Code, under Explorer> tabs > src > views > dashboards.
+      1. Create a **sampleDashboard.tsx** file.
+      1. Use the following code in the **sampleDashboard.tsx** file:
+
+        ```
+         protected dashboardLayout(): void | JSX. Element {
```

Comment:
Please change it to: `protected dashboardLayout(): JSX. Element | undefined { `

Sorry, I forgot to change the README file in the dashboard repo.

Reply:
Updated.